src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,75 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserNotValidException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    static final int MIN_LOGIN_LENGTH = 6;
+    static final int MIN_PASSWORD_LENGTH = 6;
+    static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUser(user);
+        storageDao.add(user);
+        return storageDao.get(user.getLogin());
+    }
+
+    private void checkUser(User user) {
+        if (user == null) {
+            throw new UserNotValidException("User cant be null");
+        }
+        checkLoginIsOk(user);
+        checkPasswordIsOk(user);
+        checkAge(user);
+    }
+
+    private void checkLoginIsOk(User user) {
```

Comment:
```suggestion
    private void validateLogin(User user) {
```
same for other your validate methods